pkl-core/src/main/java/org/pkl/core/runtime/StackTraceRenderer.java

```diff
@@ -101,28 +105,28 @@ private void renderSourceLine(StackFrame frame, TextFormatter out, String leftMa
             : sourceLine.length();
 
     var prefix = frame.getStartLine() + " | ";
-    out.margin(leftMargin)
-        .style(Element.LINE_NUMBER)
-        .append(prefix)
-        .style(Element.TEXT)
+    out.append(ColorTheme.STACK_TRACE_MARGIN, leftMargin)
+        .append(ColorTheme.STACK_TRACE_LINE_NUMBER, prefix)
         .append(sourceLine)
-        .newline()
-        .margin(leftMargin)
-        .repeat(prefix.length() + startColumn - 1, ' ')
-        .style(Element.ERROR)
-        .repeat(endColumn - startColumn + 1, '^')
-        .newline();
+        .appendLine()
+        .append(ColorTheme.STACK_TRACE_MARGIN, leftMargin)
+        .append(" ".repeat(prefix.length() + startColumn - 1))
+        .append(ColorTheme.STACK_TRACE_CARET, "^".repeat(endColumn - startColumn + 1))
+        .appendLine();
   }
 
-  private void renderSourceLocation(StackFrame frame, TextFormatter out, String leftMargin) {
-    out.margin(leftMargin)
-        .style(Element.TEXT)
-        .append("at ")
-        .append(frame.getMemberName() != null ? frame.getMemberName() : "<unknown>")
-        .append(" (")
-        .append(frame.getModuleUri())
-        .append(")")
-        .newline();
+  private void renderSourceLocation(
+      StackFrame frame, TextFormattingStringBuilder out, String leftMargin) {
+    out.append(ColorTheme.STACK_TRACE_MARGIN, leftMargin)
+        .append(
+            ColorTheme.STACK_FRAME,
+            () ->
+                out.append("at ")
+                    .append(frame.getMemberName() != null ? frame.getMemberName() : "<unknown>")
+                    .append(" (")
+                    .appendUntrusted(frame.getModuleUri())
```

Comment:
These values might get transformed by `~/.pkl/settings.pkl`, might inject ANSI escape codes.

Protect against that by adding reset, and re-applying all ansi codes.